File: frontend/E-commerce-Complete-Flutter-UI/lib/screens/my_account/models/user_profile_model.dart
class UserProfileModel {
  String? id;
  String? fullname;
  String? email;
  String? gender;

  UserProfileModel({this.id, this.fullname, this.email, this.gender});

  UserProfileModel.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    fullname = json['fullname'];
    email = json['email'];
    gender = json['gender'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['fullname'] = this.fullname;
    data['email'] = this.email;
    data['gender'] = this.gender;
    return data;
  }
}

File: backend/shopping-application/src/main/java/com/shopping/shoppingapplication/dtos/UserProfileModel.java
package com.shopping.shoppingapplication.dtos;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserProfileModel {
    private String id;
    private String fullname;
    private String email;
    private String gender;
}
